release-repo-scripts/bump_dependency.bash

```diff
@@ -0,0 +1,455 @@
+  git clone https://github.com/ignition-tooling/release-tools
+else
+  echo -e "${GREEN}release-tools is already cloned${DEFAULT}"
+fi
+
+cd ${TEMP_DIR}/release-tools
+git fetch
+git checkout master
+git pull
+git reset --hard
+
+for ((i = 0; i < "${#LIBRARIES[@]}"; i++)); do
+
+  LIB=${LIBRARIES[$i]}
+  VER=${VERSIONS[$i]}
+  PREV_VER="$((${VER}-1))"
+
+  ##################
+  # gazebodistro
+  ##################
+  cd ${TEMP_DIR}/gazebodistro
+
+  YAML_FILE=${LIB}${VER}.yaml
+  echo -e "${BLUE_BG}Processing [${YAML_FILE}]${DEFAULT_BG}"
+
+  if [ ! -f "${YAML_FILE}" ]; then
+    # TODO: Create it copying from version-1
+    echo -e "${RED}${YAML_FILE} does not exist.${DEFAULT}"
+    exit
+  fi
+
+  if ! grep -q main "${YAML_FILE}"; then
+    echo -e "${RED}No main branch found on ${YAML_FILE}.${DEFAULT}"
+    exit
+  fi
+
+  # Assume all files that have some `main` branch may be bumped
+  grep -rl "main" *.yaml | xargs sed -i "s ${LIB}${PREV_VER} main g"
+
+  # Add all bumped dependencies to the list to be bumped
+  DIFF=$(git diff --name-only)
+  for TO_BUMP in ${DIFF[@]}; do
+
+    TO_BUMP=${TO_BUMP%.yaml}
+    if [[ "$TO_BUMP" =~ ^([a-z-]+)([0-9]+) ]]; then
+
+      if [[ ${#BASH_REMATCH[*]} -lt 3 ]]; then
+        continue
+      fi
+
+      BUMP_LIB=${BASH_REMATCH[1]}
+      BUMP_VER=${BASH_REMATCH[2]}
+
+      if [[ ! " ${LIBRARIES[@]} " =~ " ${BUMP_LIB} " ]]; then
+        echo -e "${GREEN}Also bumping ${BUMP_LIB}${BUMP_VER}${DEFAULT}"
+        LIBRARIES+=($BUMP_LIB)
+        VERSIONS+=($BUMP_VER)
+      fi
+    fi
+
+  done
+
+  sed -i "s ${LIB}${PREV_VER} main g" collection-${COLLECTION}.yaml
+
+  ##################
+  # docs
+  ##################
+  cd ${TEMP_DIR}/docs
+  sed -i -E "s ((${LIB}.*))${PREV_VER} \1${VER} g" ${COLLECTION}/install.md
+
+done
+
+echo -e "${GREEN_BG}Commit gazebodistro? (y/n)${DEFAULT_BG}"
+read CONTINUE
+if [ "$CONTINUE" = "y" ] ; then
+  cd ${TEMP_DIR}/gazebodistro
+  git checkout -b bump_${COLLECTION}_${LIBRARY}
+  git commit -sam"Bump in ${COLLECTION}: ${LIBRARY}${VERSION}"
+  git push origin bump_${COLLECTION}_${LIBRARY}
+fi
+
+echo -e "${GREEN_BG}Commit docs? (y/n)${DEFAULT_BG}"
+read CONTINUE
+if [ "$CONTINUE" = "y" ] ; then
+  cd ${TEMP_DIR}/docs
+  git checkout -b bump_${COLLECTION}_${LIBRARY}
+  git commit -sam"Bump in ${COLLECTION}: ${LIBRARY}${VERSION}"
+  git push origin bump_${COLLECTION}_${LIBRARY}
+fi
+
+for ((i = 0; i < "${#LIBRARIES[@]}"; i++)); do
+
+  LIB=${LIBRARIES[$i]}
+  VER=${VERSIONS[$i]}
+  PREV_VER="$((${VER}-1))"
+  LIB_UPPER=`echo ${LIB#"ign-"} | tr a-z A-Z`
+
+  if ! [[ $VER == ?(-)+([0-9]) ]] ; then
+    continue
+  fi
+
+  echo -e "${BLUE_BG}Processing [${LIB}]${DEFAULT_BG}"
+
+  ##################
+  # source code
+  ##################
+
+  echo -e "${GREEN}${LIB}: source code${DEFAULT}"
+
+  cd ${TEMP_DIR}
+  if [ ! -d "${LIB}" ]; then
+    echo -e "${GREEN}Cloning ${LIB}${DEFAULT}"
+    git clone https://github.com/ignitionrobotics/${LIB}
```

Comment:
This always clones from `https://github.com/ignitionrobotics/${LIB}`, which won't work for sdformat. Can we maybe use `vcstool`?

Reply:
> Can we maybe use vcstool?

I'm open to that, the thing is that the script isn't following a yaml file. There may be a better way of doing this, but the libraries are being added to the `LIBRARIES` array recursively. Let me see if I can do something quick to support SDF.

Another assumption that the script makes is that the unreleased version is `main`, but for SDF that's `master`.

And I noticed that there's no `sdformat11-release` branch, which means we aren't making nightlies... Lots to fix, let me address some of this.

Reply:
There should be better support for SDFormat now: c0467fcd90bbb57cec33c31b00abc0677cae75e7

I didn't check it too closely though, there may be more gotchas in there.